### PR TITLE
share period resolver with dev cockpit

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/period/Period.java [hunk 1]
```diff
@@ -77,8 +78,29 @@ public long getSnapshotId() {
...
   @Override
+  public boolean equals(@Nullable Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Period period = (Period) o;
+    return index == period.index
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/period/Period.java]: MAJOR Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1067)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
LGTM

**Comment by julienlancelot** [comment]
As discusted, use selectComponentsByQualifiers instead of selectComponents then LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**